ai-catalog/samples/genai-summarization/src/main/java/com/android/ai/samples/genai_summarization/GenAISummarizationViewModel.kt

```diff
@@ -15,54 +15,75 @@
  */
 package com.android.ai.samples.genai_summarization
 
-import android.content.Context
+import android.app.Application
 import android.util.Log
-import androidx.lifecycle.ViewModel
+import androidx.annotation.StringRes
+import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.viewModelScope
 import com.android.ai.samples.geminimultimodal.R
+import com.google.mlkit.genai.common.DownloadCallback
 import com.google.mlkit.genai.common.FeatureStatus
+import com.google.mlkit.genai.common.GenAiException
 import com.google.mlkit.genai.summarization.Summarization
 import com.google.mlkit.genai.summarization.SummarizationRequest
 import com.google.mlkit.genai.summarization.Summarizer
 import com.google.mlkit.genai.summarization.SummarizerOptions
 import javax.inject.Inject
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.guava.await
 import kotlinx.coroutines.launch
 
-class GenAISummarizationViewModel @Inject constructor() : ViewModel() {
-    private val _summarizationGenerated = MutableStateFlow("")
-    val summarizationGenerated: StateFlow<String> = _summarizationGenerated
+sealed class GenAISummarizationUiState {
+    data object Initial : GenAISummarizationUiState()
+    data object CheckingFeatureStatus : GenAISummarizationUiState()
+    data class DownloadingFeature(
+        val bytesToDownload: Long,
+        val bytesDownloaded: Long,
+    ) : GenAISummarizationUiState()
 
-    private var summarizer: Summarizer? = null
+    data class Generating(val generatedOutput: String) : GenAISummarizationUiState()
+    data class Success(val generatedOutput: String) : GenAISummarizationUiState()
+    data class Error(@StringRes val errorMessageStringRes: Int) : GenAISummarizationUiState()
+}
 
-    fun summarize(textToSummarize: String, context: Context) {
-        if (textToSummarize.isEmpty()) {
-            _summarizationGenerated.value = context.getString(R.string.summarization_no_input)
-            return
-        }
+class GenAISummarizationViewModel @Inject constructor(val context: Application) : AndroidViewModel(context) {
+    private val _uiState = MutableStateFlow<GenAISummarizationUiState>(GenAISummarizationUiState.Initial)
+    val uiState: StateFlow<GenAISummarizationUiState> = _uiState.asStateFlow()
 
+    private var summarizer: Summarizer? = null
```

Comment:
You don't need the init at all, you can replace with this which will also remove the nullability

```suggestion
    private val summarizationOptions =
        SummarizerOptions.builder(context)
            .setOutputType(SummarizerOptions.OutputType.THREE_BULLETS)
            .build()
    private val summarizer: Summarizer = Summarization.getClient(summarizationOptions)
```